lib/TTT/validation.ex

```diff
@@ -35,10 +35,47 @@ defmodule Validation do
         mode
 
       _ ->
-        choose_mode(deps)
+        choose_mode(io)
     end
   end
 
+  def choose_token(io) do
+    io.retrieve(Messages.get_message(:choose_token))
+    |> clean()
+    |> validate_token(io)
+  end
+
+  def choose_token(io, player_token) do
+    io.retrieve(Messages.get_message(:choose_opponent_token))
+    |> clean()
+    |> validate_token(player_token, io)
+  end
+
+  def validate_token(token, io) when byte_size(token) != 1 do
+    io.display(Messages.get_message(:token_length_error))
+    choose_token(io)
+  end
+
+  def validate_token(token, _io) when byte_size(token) == 1 do
+    token
+  end
+
+  def validate_token(opponent_token, player_token, io) when opponent_token == player_token do
+    io.display(Messages.get_message(:same_token))
+    choose_token(io, player_token)
+  end
+
+  def validate_token(opponent_token, player_token, io)
+      when byte_size(opponent_token) != 1 do
+    io.display(Messages.get_message(:token_length_error))
+    choose_token(io, player_token)
+  end
+
+  def validate_token(opponent_token, player_token, _io)
+      when opponent_token != player_token and byte_size(opponent_token) == 1 do
```

Comment:
spacing seems weird?

Reply:
yeah it is weird, the auto-formatter thinks the guard clause is too long so it puts it on a new line